src/services/CatServices.ts

```diff
@@ -0,0 +1,45 @@
+import axios from "axios";
+import { CatList } from "../types";
+import { deepCamelCaseKeys, deepSnakeCaseKeys } from "../utils/utils";
+
+const getAllCats = async (
+  pageNumber: Number,
+  pageSize: Number
+): Promise<any> => {
+  let url = "http://localhost:10000/v1/cats?sort_by=id";
+  url = url + "&page_number=" + pageNumber;
+  url = url + "&page_size=" + pageSize;
+
+  const { data } = await axios.get(url);
+
+  const result = deepCamelCaseKeys(data);
+  return result;
+};
```

Comment:
Some points of improvement:
- Typehinting is missing (as discussed, `any` is advised and bypasses the entire reason for using TS)
- Manual building of URL is not needed when using Axios because this library will do the heave lifting for you
- You can create an axiosClient instance and re-use that in every service. This means you need to configure the API URL in only one place therefore easy maintainable and less risk of bugs
- `Number` is not a valid TypeScript type and instead should be `number` (I expect that this would even result in a console error)

```suggestion
// in types.ts
export type PagePaginatedData<ResponseDataType> = Readonly<{
  results: ReadonlyArray<ResponseDataType>;
  metadata: Readonly<{
    hasNextPage: boolean;
  }>;
}>;

export type CatSummary = {
  id: string;
  name: string;
  url: string | null;
};

export type CatSummaryList = PagePaginatedData<CatSummary>;


// In axiosClient.ts
const API_URL = "http://localhost:10000/v1/";
const TIMEOUT_IN_MS = 5 * 60 * 1000;

const axiosClient = axios.create({
  timeout: TIMEOUT_IN_MS,
  baseURL: API_URL,
});

// In CatService.ts, with everything in place, it is very trivial now to make a API call
const getAllCats = async (
  pageNumber: number,
  pageSize: number
): Promise<CatSummaryList> => {
  const params = deepSnakeCaseKeys({
    pageSize,
    pageNumber,
  });

  const { data } = await axiosClient.get("/cats", { params });

  return deepCamelCaseKeys(data);
};
```
WDYT?